src/test/java/fi/aalto/cs/apluscourses/intellij/actions/OpenItemActionTest.java

```diff
@@ -15,80 +15,127 @@
 import fi.aalto.cs.apluscourses.intellij.notifications.SubmissionRenderingErrorNotification;
 import fi.aalto.cs.apluscourses.intellij.services.MainViewModelProvider;
 import fi.aalto.cs.apluscourses.model.Exercise;
+import fi.aalto.cs.apluscourses.model.ExerciseGroup;
 import fi.aalto.cs.apluscourses.model.SubmissionResult;
 import fi.aalto.cs.apluscourses.model.UrlRenderer;
 import fi.aalto.cs.apluscourses.presentation.MainViewModel;
-import fi.aalto.cs.apluscourses.presentation.base.BaseTreeViewModel;
+import fi.aalto.cs.apluscourses.presentation.base.SelectableNodeViewModel;
 import fi.aalto.cs.apluscourses.presentation.exercise.ExerciseGroupViewModel;
 import fi.aalto.cs.apluscourses.presentation.exercise.ExerciseViewModel;
 import fi.aalto.cs.apluscourses.presentation.exercise.ExercisesTreeViewModel;
 import fi.aalto.cs.apluscourses.presentation.exercise.SubmissionResultViewModel;
 import fi.aalto.cs.apluscourses.presentation.filter.Options;
-import org.junit.Before;
+import java.util.Collections;
 import org.junit.Test;
 import org.mockito.ArgumentCaptor;
 
-public class OpenSubmissionActionTest {
+public class OpenItemActionTest {
 
+  private final String submissionString = "submission";
+  private final String exerciseString = "exercise";
+  private final String weekString = "week";
+  private Exercise exercise;
   private SubmissionResult submissionResult;
+  private ExerciseGroup exerciseGroup;
   private AnActionEvent actionEvent;
   private MainViewModelProvider mainViewModelProvider;
   private Notifier notifier;
-  private UrlRenderer submissionRenderer;
+  private UrlRenderer urlRenderer;
 
   /**
    * Called before each test.
+   * @param viewModelType The type of SelectableNodeViewModel to be tested, as a string
    */
-  @Before
-  public void setUp() {
-    Exercise exercise = new Exercise(223, "TestEx", "http://example.com", 0, 1, 10, true);
+  public void setUp(String viewModelType) {
```

Comment:
I recommend passing an instance of `SelectableNodeViewModel` directly instead of the current approach which, as you noted, isn't that nice. So one test would create an exercise group view model and pass that, another would create a submission result view model, and so on

Reply:
oh yeah why didn't I do that